### PR TITLE
refactor: improve get identity path duplicate code

### PR DESCRIPTION
**Pull request by BlackHole1**
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/util/utils.go [hunk 1]
```diff
@@ -473,17 +474,8 @@ func ExitCode(err error) int {
...
-func HomeDir() (string, error) {
-	home := os.Getenv("HOME")
-	if home == "" {
-		usr, err := user.LookupId(fmt.Sprintf("%d", rootless.GetRootlessUID()))
-		if err != nil {
-			return "", fmt.Errorf("unable to resolve HOME directory: %w", err)
-		}
-		home = usr.HomeDir
-	}
-	return home, nil
```
Comment (BlackHole1) [inline pkg/util/utils.go:486]: It is invalid code. Because moved to `containers/storage/pkg/unshare.HomeDir`

File: pkg/machine/keys.go [hunk 2]
```diff
@@ -33,18 +33,16 @@ func CreateSSHKeys(writeLocation string) (string, error) {
 	return strings.TrimSuffix(string(b), "\n"), nil
 }
 
-func CreateSSHKeysPrefix(dir string, file string, passThru bool, skipExisting bool, prefix ...string) (string, error) {
-	location := filepath.Join(dir, file)
-
-	_, e := os.Stat(location)
+func CreateSSHKeysPrefix(identityPath string, passThru bool, skipExisting bool, prefix ...string) (string, error) {
```
Comment (TomSweeneyRedHat) [inline pkg/machine/keys.go:36]: Does this need to be public?

File: pkg/machine/keys.go [hunk 3]
```diff
@@ -74,7 +72,14 @@ func generatekeys(writeLocation string) error {
 }
 
 // generatekeys creates an ed25519 set of keys
-func generatekeysPrefix(dir string, file string, passThru bool, prefix ...string) error {
+func generatekeysPrefix(identityPath string, passThru bool, prefix ...string) error {
+	dir := filepath.Dir(identityPath)
+	file := filepath.Base(identityPath)
```
Comment (TomSweeneyRedHat) [inline pkg/machine/keys.go:77]: I think we need to do some additional checks here of the values returned.  First off, if `file` is blank or is a `.`, I think we should fail.

More importantly, if I manage to pass "../../etc/dev.txt", I'm going to be writing potentially in etc or other places I should not.  Is this a security concern?  Hopefully I'm just misreading the code.

File: pkg/machine/keys.go (hunk 2 above)
Comment (BlackHole1) [reply pkg/machine/keys.go:36]: Yes. The reference can be found at: https://github.com/containers/podman/blob/249f0463ebf8fa1cc82fd4d345aca2f244cf5a5d/pkg/machine/wsl/machine.go#L975-L977

File: pkg/machine/keys.go (hunk 3 above)
Comment (BlackHole1) [reply pkg/machine/keys.go:77]: As far as I can see, there shouldn't be any security issues. In fact, the modification to the function parameter is very simple: previously, two parameters were concatenated within the function, but now the concatenation operation has simply been moved to the location where the function is called. There is no change in security compared to before. At the same time, I have rechecked the function call chain, and this function has a unique calling path. Ultimately, the function is called by the Init function in `pkg/machine/wsl/machine.go`, with the first parameter defined here: https://github.com/containers/podman/blob/249f0463ebf8fa1cc82fd4d345aca2f244cf5a5d/pkg/machine/wsl/machine.go#L410-L411.

Comment (BlackHole1) [reply pkg/machine/keys.go:77]: Currently, this PR only refactors some code to make it more maintainable, and the security indicators will be exactly the same as before. To increase reliability, I will be asynchronously testing some security issues, but this is beyond the scope of this PR. If there are any security issues found, I will submit a new PR for the fixes. 😃

**Comment by BlackHole1** [comment]
Request CR again.
/CC @TomSweeneyRedHat

**Comment by TomSweeneyRedHat** [comment]
@BlackHole1, thanks for the explanations on my comments.  However, I don't have the time to dig deep at the moment, and I don't feel good about adding my own LGTM to this.  If others feel more comfortable doing so, I will not hold up the PR.

**Comment by BlackHole1** [comment]
@TomSweeneyRedHat OK, thank you equally for your CR :)

File: pkg/util/utils_test.go [hunk 4]
```diff
@@ -2,9 +2,11 @@ package util
 
 import (
 	"fmt"
+	"path/filepath"
 	"testing"
 	"time"
 
+	"github.com/docker/docker/pkg/homedir"
```
Comment (rhatdan) [inline pkg/util/utils_test.go:9]: Why not containers/storagek/pkg/homedir

**Comment by rhatdan** [comment]
@ashley-cui @cdoern PTAL

File: pkg/util/utils_test.go (hunk 4 above)
Comment (ashley-cui) [reply pkg/util/utils_test.go:9]: +1, this test does not test the correct package/code change

Comment (BlackHole1) [reply pkg/util/utils_test.go]: Wow. It looks like I selected the wrong path when using auto import in `Goland` 😆. It has now been corrected ✅.

**Comment by rhatdan** [comment]
/approve
/lgtm
Thanks @BlackHole1

**Pull request merged**